travis-ci/vmtest/run.sh

```diff
@@ -378,10 +378,17 @@ LIBBPF_PATH="${REPO_ROOT}" \
 	VMTEST_ROOT="${VMTEST_ROOT}" \
 	VMLINUX_BTF=${vmlinux} ${VMTEST_ROOT}/build_selftests.sh
 
+declare -A tests_results
+
 travis_fold start bpftool_checks "Running bpftool checks..."
 if [[ "${KERNEL}" = 'LATEST' ]]; then
-	"${REPO_ROOT}/tools/testing/selftests/bpf/test_bpftool_synctypes.py" && \
-		echo "Consistency checks passed successfully."
+	"${REPO_ROOT}/tools/testing/selftests/bpf/test_bpftool_synctypes.py" && true
```

Comment:
does `&& true` changes anything? Doesn't it overrides the exit code? I suspect I'm missing some subtle behavior here, can you please explain?

Reply:
It does not change the exit code, because if the Python script fails, the `echo true` is never evaluated.

But it does have an effect: Because we have a `trap 'exit 2' ERR` at the beginning of the file, `run.sh` would exit if the Python script failed. Adding the `&& true` prevents that. Here's what the man page for bash says:

> The `ERR` trap is not executed if the failed command is [...] part of a command executed in a `&&` or `||` list except the command following the final `&&` or `||`, [...].

I'll add a comment to make it clear.